web/app/js/components/BaseTable.test.js

```diff
@@ -82,4 +82,42 @@ describe("Tests for <BaseTable>", () => {
     expect(emptyCard).toBeDefined();
     expect(emptyCard).toHaveLength(1);
   });
+
+  it("filters the table", () => {
+    let extraProps = _merge({}, defaultProps, {
+      tableRows: [{
+        deployment: "authors",
+        namespace: "default",
+        key: "default-deployment-authors",
+        pods: {totalPods: "1", meshedPods: "1"}
+      },
+      {
+        deployment: "books",
+        namespace: "default",
+        key: "default-deployment-books",
+        pods: {totalPods: "2", meshedPods: "1"}
+      }],
+      tableColumns: [{
+        dataIndex: "deployment",
+        title: "Name",
+        filter: x => x.deployment,
+      },
+      {
+        dataIndex: "namespace",
+        title: "Namespace"
+      }],
+      enableFilter: true
+    });
+
+    const component = mount(routerWrap(BaseTable, extraProps));
+
+    expect(component.find("TableBody").find("TableRow")).toHaveLength(2);
+
+    component.find("BaseTable").instance().setState({
+      filterBy: new RegExp(/auth/),
```

Comment:
This is a tiny nit but given that we use the term `auth` in so many different ways in the Linkerd repo and coding in general, I would change this to `authors` or even `books` so it's clear that this is a search term.